libmime/patch.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2018 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libmime
+
+import "mime"
+
+// Overrider defines a type that overrides a <ext, mimeType> mapping.
+type Overrider func(ext string, mimeType string) (newExt string, newMimeType string)
+
+func dontOverride(ext string, mimeType string) (newExt string, newMimeType string) {
+	return ext, mimeType
+}
+
+// Patch patches the mime types Go uses by calling mime.AddExtensionType on
+// each from a private list in this package. Optionally provide a non-nil
+// Overrider to override any mime type defined in the list.
+func Patch(override Overrider) {
```

Comment:
Is this basically a required init function for anything using mime types?  If so, this seems like a weird function name, and something like `Init()` would be better.

Or does "mime" already have most of these defaults, and most of these `AddExtensionType` calls are no-ops?

Reply:
`mime` loads from the a couple places on the filesystem. So if the machine has nginx or apache installed, `mime` can already use them. But in many cases these are not no-ops. Also for `html` ones, we are overriding them into plain text to avoid being executed by frontend web view components.

`Init()` sounds good!

Reply:
Actually I think patch is still better, based on your comment. Let me know if you still want `Init`